echo_server.py

```diff
@@ -45,15 +53,59 @@ def echo_server():
     server_socket.close()
 
 
-def response_ok(body=''):
+def resolve_uri(uri):
+    """Given a URI, return a body and content type.
+
+    If the URI is a directory, return that directory as a list as the body.
+    If the URI a file, return the contents of the file as the body
+    Content type is the file type.
+    Raises type error if uri is not a string.
+    Raises an IO error if the uri is not a file or directory.
+    Raises an IO error if cannot find a mimetype.
+    """
+    if not isinstance(uri, str):
+        raise HTTPError404('Not found')
+    uri = os.path.join(ROOT_DIR, uri)
+    extension = os.path.splitext(uri)
+    try:
+        content_type = mimetypes.types_map[extension[1]]
+    except KeyError:
+        content_type = ''
```

Comment:
probably better to default to something like `text/plain` rather than an empty string.